common/unsafe/src/main/java/org/apache/spark/unsafe/hash/Murmur3_x86_32.java

```diff
@@ -87,6 +106,35 @@ public static int hashUnsafeBytes2(Object base, long offset, int lengthInBytes,
     return fmix(h1, lengthInBytes);
   }
 
+  public static int hashUnsafeBytes2Block(MemoryBlock base, int seed) {
```

Comment:
Can't we just call `hashUnsafeBytes2` with the `baseObject`, `baseOffset` and `size` of `base`?

Reply:
ditto